src/components/views/settings/ChangePassword.tsx

```diff
@@ -307,11 +320,11 @@ export default class ChangePassword extends React.Component<IProps, IState> {
         const oldPassword = this.state.oldPassword;
         const newPassword = this.state.newPassword;
         const confirmPassword = this.state.newPasswordConfirm;
-        const err = this.checkPassword(oldPassword, newPassword, confirmPassword);
-        if (err) {
-            this.props.onError(err);
-        } else {
+        try {
+            this.checkPassword(oldPassword, newPassword, confirmPassword);
```

Comment:
Potentially, we don't even need this function if we trust our `PassphraseField.tsx` to do the right thing. Both of these cases are covered by the logic there and since we `verifyFieldsBeforeSubmit()`  just above there is no way `checkPassword` would ever throw one of these errors since they are already valid.


<img width="596" alt="Field validation: Passwords can't be empty" src="https://user-images.githubusercontent.com/558581/232179902-1716686a-aea2-49cf-99d3-bf2c64d4cfa9.png">
<img width="591" alt="Field validation: Passwords don't match" src="https://user-images.githubusercontent.com/558581/232179901-f5169bcc-3551-4f7b-9d36-9c67a5f4694d.png">
<img width="570" alt="Field validation when confirm password is empty: Confirm password" src="https://user-images.githubusercontent.com/558581/232179903-440c022e-6494-4d45-89b7-45bcbb99b2dc.png">

Reply:
Removed ⏩